File: 0019 删除链表的倒数第N个节点/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        length=0
        node=head
        LOfN=[]
        while(node):
            LOfN.append(node)
            node = node.next
        if(n>=len(LOfN)):
            head=head.next
        else:

            LOfN[-n-1].next=LOfN[-n].next
        return head
            

File: 0019 删除链表的倒数第N个节点/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode hhead(0,head);
        ListNode* qi=&hhead,*si=&hhead;
        for(int i=0;i<n;i++){
            qi=qi->next;
        }
        while(qi->next != nullptr){
            qi=qi->next;
            si=si->next;
        }
        si->next=si->next->next;
        return hhead.next;
    }
};